ibm/mas_devops/roles/db2/templates/tlsroute.yml.j2

```diff
@@ -1,7 +1,7 @@
 kind: Route
 apiVersion: route.openshift.io/v1
 metadata:
-  name: db2u-tls-route
+  name: db2u-{{ db2_instance_name }}-tls-route
```

Comment:
We need to put cleanup logic into the role to remove the old route, otherwise when running an over-the-top install, user will be left with two routes that conflict.

- If possible we delete the old route after creating the new one, **but we need to test whether the two routes can co-exist**  If they cannot co-exist we need to delete the old route first, and accept a brief outage.

Reply:
I think it's okay to have two routes pointing to the same service, but I'll test and get back to you.

Reply:
Glad you mentioned this... indeed i get an issue with code `HostAlreadyClaimed ` if having two routes exposing the same endpoint location.

<img width="639" alt="image" src="https://user-images.githubusercontent.com/31037381/200667009-d0082aa1-9fef-4a93-a948-52f43139ef79.png">

Good thing is that in a scenario with two routes pointing to same host/endpoint, while existing route keeps active and serving the requests, the new created one conflicts and goes into error state. So, deleting old route replaces the endpoint automatically to the newly created route, with no downtime.

<img width="737" alt="image" src="https://user-images.githubusercontent.com/31037381/200667776-68850c2e-6e8a-4b3c-85b7-62fb3527b52a.png">

Therefore, as you suggested, i implemented the clean up step to remove old hardcoded `db2u-tls-route` after creating the new custom route, and only if the host/endpoint conflicts between old and new routes. That way we ensure that we only clean up the route in a conflicting host scenario.